### PR TITLE
Fix false positive for `Style/EmptyLineAfterGuardClause`

### PR DESCRIPTION
**Pull request by koic**
Related #5679.

This PR fixes a false positive for Style/EmptyLineAfterGuardClause when guard clause is before `else`.

## Reproduction code

```ruby
def foo
  if cond
    return another_object if something_different?
  else
    bar
  end
end
```

## Expected behavior

No offenses.

## Actual behavior and Steps to reproduce the problem

```console
% rubocop /tmp/example.rb --only Style/EmptyLineAfterGuardClause
Inspecting 1 file
C

Offenses:

/tmp/example.rb:3:5: C: Style/EmptyLineAfterGuardClause: Add empty line
after guard clause.
    return another_object if something_different?
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```console
% rubocop -V
0.53.0 (using Parser 2.5.0.4, running on ruby 2.5.0 x86_64-darwin17)
```

This PR fixes the above false positive.

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [ ] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/style/empty_line_after_guard_clause.rb [hunk 1]
```diff
@@ -73,6 +74,15 @@ def next_line_rescue_or_ensure?(node)
           parent.nil? || parent.rescue_type? || parent.ensure_type?
         end
 
+        def next_sibling_parent_empty_or_else?(node)
+          next_sibling = node.parent.children[node.sibling_index + 1]
+          return true if next_sibling.nil?
```
Comment (Drenmi) [inline lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: Why is this cop `Style/EmptyLineAfterGuardClause` and not `Style/EmptyLinesAroundGuardClause`. 😕

Comment (koic) [reply lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: This cop is responsible for checking that there is a blank line after guard clause (not around), so I think that this name is good 😃 
cc/ @unkmas

Comment (unkmas) [reply lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: @koic :+1:

Comment (Drenmi) [reply lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: I'm not saying the name is wrong, I'm saying that only checking after the guard clause seems like only "half a cop." 🙂

Comment (unkmas) [reply lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: @Drenmi I don't think, that before guard clause always should be a blank line.

```ruby
def foo
  result = SomeAction.peform
  return 'WTF' if result.failure?

  result.payload
end
```

This piece of code feels very natural for me.

Comment (Drenmi) [reply lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: @unkmas: That's fine. It should be configurable to everyone's own preference. But by naming the cop like this, I can't add configuration options to it. 🙂 I would need to add another, `Style/EmpyLineBeforeGuardClause` that duplicates the implementation.

Comment (koic) [reply lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: I see. I understand about the question of `Style/EmptyLinesAroundGuardClause`.
However, I think that the style of putting a blank line before guard clause feel a bit iffy 💦
In the case of this cop, I think it is important to focus only on after of guard clause.

Comment (Drenmi) [reply lib/rubocop/cop/style/empty_line_after_guard_clause.rb]: Well. There are as many "feelings" about code as there are developers, which is why the vision of RuboCop has shifted towards being fully configurable. 🙂

**Pull request merged**